docs/pages/access-controls/guides/impersonation.mdx

```diff
@@ -0,0 +1,301 @@
+
+- Alice can get a new `jenkins` certificate with the same TTL and some fields updated.
+Teleport allows impersonated users to renew their certificates with with reduced the scope of the certificate.
+
+## Step 3/3 Dynamic Impersonation
+
+Sometimes you don't know in advance what roles will be created by the system.
+
+You can use `where` condition to allow one role to impersonate other roles
+based on matching labels.
+
+Save the role `security-impersonator` in a file `security-impersonator.yaml` and create it using `tctl`.
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: security-impersonator
+spec:
+  options:
+    max_session_ttl: 10h
+
+  # Allow security-impersonator to impersonate any user and role with group: security label
+  allow:
+    impersonate:
+      users: ['*']
+      roles: ['*']
+      where: 'equals(impersonate_role.metadata.labels["group"], "security") && equals(impersonate_user.metadata.labels["group"], "security")'
+
+  # the deny section uses the identical format as the 'allow' section.
+  # the deny rules always override allow rules.
+  deny:
+    node_labels:
+      '*': '*'
+```
+
+```bash
+$ tctl create -f security-impersonator.yaml
+$ tctl users update alice --set-roles=security-impersonator,access
+```
+
+Alice can now impersonate any role and user with a label `group: security`.
+
+Create a user and a role `security-scanner`. Save a file `security-scanner.yaml` and create user and group:
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: scanner
+  labels:
+    group: security
+spec:
+  options:
+    max_session_ttl: 10h
+
+  allow:
+    logins: ['root']
+    node_labels:
+      '*': '*'
+---
+kind: user
+version: v2
+metadata:
+  name: security-scanner
+  labels:
+    group: security
+spec:
+  roles: ['security-scanner']
+```
+
+Alice can issue certificates for a security scanner:
+
+``` bash
+$ tctl auth sign --user=security-scanner --format=openssh --out security-scanner --ttl=10h
+```
+
+**Matching user traits**
+
+We can make security-impersonator role more flexible by matching source user and target role parameters:
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: security-impersonator
+spec:
+  options:
+    max_session_ttl: 10h
+
+  allow:
+    impersonate:
+      users: ['*']
+      roles: ['*']
+      where: 'contains(user.spec.traits["group"], impersonate_role.metadata.labels["group"]) && contains(user.spec.traits["group"], impersonate_user.metadata.labels["group"])'
+
+  # the deny section uses the identical format as the 'allow' section.
+  # the deny rules always override allow rules.
+  deny:
+    node_labels:
+      '*': '*'
+```
+
+This role will work with SSO users, but we can test locally, by updating Alice's account
+entry with traits:
+
+```yaml
+kind: user
+version: v2
+metadata:
+  name: alice
+spec:
+  traits:
+    group: ['security', 'devops']
+  roles:
+  - security-impersonator
+  - access
+```
+
+```bash
```

Comment:
Also, this should probably show how to actually update the local `alice` user with those traits (the command).